tests/gbs/test_similarity.py

```diff
@@ -194,3 +196,105 @@ def test_sample_max_count_deterministic(self, photon_num, count, monkeypatch):
             m.setattr("numpy.random.choice", lambda x: x[0])
             samp = similarity.event_to_sample(photon_num, count, modes_dim)
         assert samp[0] == count
+
+
+def test_orbit_cardinality():
+    """Test if function ``strawberryfields.gbs.similarity.orbit_cardinality`` returns the
+    correct number of samples for some hard-coded examples."""
+
+    orbits = {
+        ((1, 1, 2), 4): 12,
+        ((1, 1), 4): 6,
+        ((1, 2, 3), 4): 24,
+        ((1, 1, 1, 1), 5): 5,
+        ((1, 1, 2), 5): 30,
+        ((1, 2, 3), 5): 60,
+    }
+
+    calc_cardinalities = {}
+    for o, _ in orbits.items():
+        calc_cardinalities[o] = similarity.orbit_cardinality(*o)
+
+    assert calc_cardinalities == orbits
+
+
+def test_event_cardinality():
+    """Test if function ``strawberryfields.gbs.similarity.event_cardinality`` returns the
+    correct number of samples for some hard-coded examples."""
+
+    events = {
+        (5, 3, 6): 216,
+        (6, 3, 6): 336,
+        (5, 2, 6): 126,
+        (5, 3, 7): 413,
+        (6, 3, 7): 728,
+        (5, 2, 7): 266,
+    }
+
+    calc_cardinalities = {}
+    for o, _ in events.items():
+        calc_cardinalities[o] = similarity.event_cardinality(*o)
+
+    assert calc_cardinalities == events
+
+
+class TestProbOrbitMC:
+    """Tests for the function ``strawberryfields.gbs.similarity.p_orbit_mc.``"""
+
+    @pytest.mark.parametrize("n_mean", [0, 2, 4])
+    def test_actual_prob_orbit(self, n_mean):
+        """Tests if the output of the function is a valid probability between 0 and 1."""
+        graph = nx.complete_graph(10)
+
+        assert 0 <= similarity.p_orbit_mc(graph, [], n_mean) <= 1
```

Comment:
I'm quite nervous about this non-deterministic test 😅 
Isn't it possible to exceed 1 if we hit some high probability events?

Reply:
Hmmm, for the orbit `[]` the output will be the probability of vacuum, which will not exceed 1. However, you still have a good point: maybe it is a good idea to monkeypatch all relevant non-deterministic parts of the function?